client/cody/src/chat/ChatViewProvider.ts

```diff
@@ -160,11 +160,20 @@ export class ChatViewProvider implements vscode.WebviewViewProvider {
                     try {
                         const doc = await vscode.workspace.openTextDocument(uri)
                         await vscode.window.showTextDocument(doc)
-                    } catch (error) {
-                        console.error(`Could not open file: ${error}`)
+                    } catch  {
+                        // Try to open the file in the sourcegraph view
+                        const sourcegraphInstanceUrl = 'https://sourcegraph.com'
+                        const sourcegraphWebUrl = `${sourcegraphInstanceUrl}/search?q=context:global+file:${message.filePath}`
```

Comment:
```suggestion
                        const sourcegraphWebUrl = new URL(`/search?q=context:global+file:${message.filePath}`, sourcegraphInstanceUrl).href
```

Suggest using URL constructor to avoid https://github.com/sourcegraph/sourcegraph/issues/50186

Reply:
@abeatrix I have made the changes.